src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -1120,17 +1115,21 @@ void Syscollector::syncLoop()
 
 void Syscollector::push(const std::string& data)
 {
-    auto rawData{data};
-    Utils::replaceFirst(rawData, "dbsync ", "");
-    const auto buff{reinterpret_cast<const uint8_t*>(rawData.c_str())};
-    try
-    {
-        m_spRsync->pushMessage(std::vector<uint8_t>{buff, buff + rawData.size()});
-    }
-    // LCOV_EXCL_START
-    catch(const std::exception& ex)
+    std::unique_lock<std::mutex> lock{m_mutex};
```

Comment:
why is the lock needed here?

Reply:
This is because you can try to access m_spRsync from another thread, (the one with the push) when the shutdown process has already been done and m_spRsync has been cleaned.

Reply:
![image](https://user-images.githubusercontent.com/14300208/102620014-1dd01d00-411c-11eb-8d0b-3f93bf25e9ac.png)